### ISSUE TITLE
issue: There is a problem retrieving data using the shouldUnregister configuration 

### ISSUE THREAD

**Issue by ibeizhu** [react-hook-form/react-hook-form#12490]
### Version Number

7.54.1

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/inspiring-johnson-j4zrpx?file=%2Fsrc%2FApp.js%3A25%2C14&workspaceId=ws_DUpBMQkanMVMH4sxTkz7fG

### Steps to reproduce

useForm with shouldUnregister can handle automatic clearing of data under linked logic, but it results in a poor user experience.

For example, I have four radio buttons A, B, C, and D that can toggle types, each associated with different fields.

When shouldUnregister = true, if the user fills in the fields under type A and then switches to type B, returning to type A results in the previously filled data being completely cleared, which is very unfriendly to the user.

When shouldUnregister = false, this problem can be resolved, but if users fill in data for both types A and B, when submitting the form, while using form.getValues(), the data will include dirty data from type B since type A is the final submission, requiring developers to manually clear this. This is also unfriendly for developers since the UI logic needs to reprocess the data upon submission, which is prone to errors.

### Expected behaviour

Is it possible to provide a new parameter shouldUnregister to the getValues function? This would allow users to obtain clean form data by calling form.getValues(shouldUnregister), automatically excluding hidden data.

In other words, when shouldUnregister = false, it ensures a good user experience for UI editing. When retrieving data, by using form.getValues({ shouldUnregister: true }), it only fetches the data from registered fields, ensuring the accuracy of the submitted data.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ibeizhu** [comment]
Can any official developer respond to whether my question and suggestion will be addressed? thank you.

**Comment by IchabodCrow** [comment]
I agree with @ibeizhu. I have a big problem with development. I can't write code without pain. Every time I add a new dynamic field to the form, I have to change the submit handler. This is very inconvenient to use. Really looking forward to hearing back on this issue

**Comment by ibeizhu** [comment]
@IchabodCrow @evocateur @aaronjensen @tonycoco As the official developers have not yet replied to this issue, I have opened a new discussion. Please assist by upvoting it to ensure the officials notice it.

[https://github.com/orgs/react-hook-form/discussions/12666](https://github.com/orgs/react-hook-form/discussions/12666)

**Comment by PraveenPeri** [comment]
@bluebill1049 Could you clarify that with `Controller` component, there is no way currently to say `keepValue: true` during unmounting? `getValues` behavior changed in the recent versions and we want it to retain form state of the unmounted inputs. Wondering how to do that with `Controller` component

**Comment by bluebill1049** [comment]
Thanks for the PR. I think this is more of a feature request than a bug/issue — at the moment we probably don’t want to bake browser-specific behavior directly into the library.
